packages/flutter/lib/src/material/pickers/calendar_date_picker.dart

```diff
@@ -555,13 +612,98 @@ class _MonthPickerState extends State<_MonthPicker> {
     );
   }
 
+  /// Handler for when the overall day grid obtains or loses focus.
+  void _handleGridFocusChange(bool focused) {
+    setState(() {
+      if (focused) {
+        if (_focusedDay == null) {
```

Comment:
Could combine these if statements

```suggestion
      if (focused && _focusedDay == null) {
```